ggircs-app/app/pages/xml-diff.tsx

```diff
@@ -1,102 +1,60 @@
 import React, { Component } from "react";
 import { graphql } from "react-relay";
 import { Row, Col } from "react-bootstrap";
-import Button from "@button-inc/bcgov-theme/Button";
 import { xmlDiffQueryResponse } from "xmlDiffQuery.graphql";
 import { PageComponentProps } from "next-env";
 import DefaultLayout from "components/Layout/DefaultLayout";
 import { USER_GROUP } from "data/group-constants";
 import ReportSelector from "components/XmlDiff/ReportSelector";
-import DiffDetails from "components/XmlDiff/DiffDetails";
-import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { faExchangeAlt } from "@fortawesome/free-solid-svg-icons";
-import format from "xml-formatter";
+import DiffDetailsContainer from "components/XmlDiff/DiffDetailsContainer";
 import RenderDiff from "components/XmlDiff/RenderDiff";
-// Use dynamic import for BootstrapSwitchButton (workaround for SSR bug where window is not defined)
-import dynamic from "next/dynamic";
-const BootstrapSwitchButton = dynamic(
-  () => import("bootstrap-switch-button-react"),
-  {
-    loading: () => <p>loading</p>,
-    ssr: false, // This line is important. It's what prevents server-side render
-  }
-);
+import DiffControls from "components/XmlDiff/DiffControls";
 
 const ALLOWED_GROUPS = [...USER_GROUP];
 
 interface Props extends PageComponentProps {
   query: xmlDiffQueryResponse["query"];
 }
-interface State {
-  leftSideId: Number;
-  leftSideReport: any;
-  rightSideId: Number;
-  rightSideReport: any;
-  isReversed: Boolean;
-  isCollapsed: Boolean;
-  renderDiff: Boolean;
-}
-export default class Index extends Component<Props, State> {
+
+export default class Index extends Component<Props> {
   static allowedGroups = ALLOWED_GROUPS;
   static isAccessProtected = true;
   static query = graphql`
-    query xmlDiffQuery {
+    query xmlDiffQuery($FirstSideRelayId: ID!, $SecondSideRelayId: ID!) {
       query {
         session {
           ...DefaultLayout_session
         }
         allReports {
           edges {
             node {
+              id
               swrsReportId
-              latestSwrsReport {
-                submissionDate
-                ecccXmlFileByEcccXmlFileId {
-                  xmlFileName
-                  xmlFile
-                  ecccZipFileByZipFileId {
-                    zipFileName
-                  }
-                }
-              }
             }
           }
         }
```

Comment:
Maybe as a tech debt, we could avoid pulling the full list of reports and only verify if a report exists when the user types it in

Reply:
captured in issue #344